sys/vm/swap_pager.c

```diff
@@ -150,11 +150,14 @@ __FBSDID("$FreeBSD$");
 #endif
 
 #define	SWAP_META_PAGES		PCTRIE_COUNT
+
 #if __has_feature(capabilities)
-#define	BITS_PER_TAGS_PER_PAGE					\
-	((PAGE_SIZE / CHERICAP_SIZE) / (8 * sizeof(uint64_t)))
+typedef u_long tag_word_t;
```

Comment:
I might have just used `u_long` directly rather than a typedef.  It makes it more obvious that it's ok to use `ffsl` (the original motivation for my suggestion) and you can also use `1ul` in the shift instead of a cast.

Reply:
You could... my rationale was that, once everything is u_long, it can be hard to change the type of something everywhere. But it's true that not much really cares about the type now, outside of the definitions here it's just swp_pager_meta_cheri_get_tags that uses the type, and even then it's assuming it's a long by using ffsl.

Reply:
Added a tag_word_ffs instead. Maybe a little overkill, but less effort than killing off tag_word_t